lab_2_tokenize_by_bpe/main.py

```diff
@@ -26,6 +37,18 @@ def collect_frequencies(
     :param end_of_word: a token that signifies the end of word
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if not isinstance(text, str) or (start_of_word is not None and not isinstance(start_of_word, str)) \
+            or not isinstance(end_of_word, str):
+        return None
+    words = text.split()
+    frequency_dict = {}
+    for word in words:
+        prepared_word = prepare_word(word, start_of_word, end_of_word)
+        if prepared_word is not None:
+            frequency_dict[prepared_word] = frequency_dict.get(prepared_word, 0) + 1
+        elif prepared_word is None:
+            return None
```

Comment:
1. `elif` here could be replaced by `else`, do you agree?
2. make it an early return: if the output value is `None`, return `None` immediately and save indentation